### PR TITLE
Extract ASGI scope creation into function

### PR DESCRIPTION
**Pull request by emcpow2**
* Simplify magnum call function
* extract event processing into dedicated functions

This is a required abstraction to create non-AWS event handlers.

### CODE REVIEW

**Comment by jordaneremieff** [comment]
Thanks for the PR @emcpow2. There is an issue [here](https://github.com/jordaneremieff/mangum/issues/147) related to custom scope resolution that should also be considered when making a change like this, and I think if we decide to go with non-AWS event handling that will take some additional design consideration later as well.

I'm going to close this because there is some investigation to do before we start extracting logic from the main adapter method.

**Comment by emcpow2** [comment]
Thit PR does not introduce any complexity regarding the path resolution issue.

Instead, the code looks cleaner and easier to test and to introduce new changes.

Please consider reopening the PR.

**Comment by jordaneremieff** [comment]
@emcpow2 I don't disagree with your approach generally, and I've actually thought we should do something _like_ this for awhile now. However, I've held off on these kind of changes because once we introduce new public API then users will start relying on it in ways that conflict with other things we might want to do. This the same reason I left the linked issue open because the design needs careful consideration.

**Comment by jordaneremieff** [comment]
@emcpow2 I've taken the time to think about this some more - let's go ahead it with it. I'll reopen and do a small review of some adjustments, but otherwise this should be fine to do now. Thanks.

File: mangum/adapter.py [hunk 1]
```diff
@@ -91,89 +91,108 @@ def __call__(self, event: dict, context: "LambdaContext") -> dict:
...
-            if event.get("version") == "2.0":
-                source_ip = request_context["http"]["sourceIp"]
-                path = request_context["http"]["path"]
-                http_method = request_context["http"]["method"]
-                query_string = event.get("rawQueryString", "").encode()
+        return response
 
-                if event.get("cookies"):
-                    headers["cookie"] = "; ".join(event.get("cookies", []))
+    def event_to_request_body(self, event: dict) -> bytes:
```
Comment (jordaneremieff) [inline mangum/adapter.py:104]: Remove this method and keep the logic in the main `__call__` function for now.

File: mangum/adapter.py [hunk 2]
```diff
@@ -91,89 +91,108 @@ def __call__(self, event: dict, context: "LambdaContext") -> dict:
...
-                "scheme": headers.get("x-forwarded-proto", "https"),
-                "query_string": query_string,
-                "server": server,
-                "client": client,
-                "asgi": {"version": "3.0"},
-                "aws.event": event,
-                "aws.context": context,
+        return initial_body
+
+    def event_to_asgi_scope(self, event: dict) -> Scope:
```
Comment (jordaneremieff) [inline mangum/adapter.py:115]: Name this `create_scope`.

**Review by jordaneremieff — COMMENTED** [review]
Just a few small things then I think this will be good to merge. The most significant change request is to remove the separate method for handling the initial body (we can look at this as a separate issue later).

File: mangum/adapter.py (hunk 2 above)
Comment (emcpow2) [reply mangum/adapter.py]: done. Since that's a private method - renamed to `_create_scope`

File: mangum/adapter.py (hunk 1 above)
Comment (emcpow2) [reply mangum/adapter.py]: done

File: mangum/adapter.py [hunk 3]
```diff
@@ -91,89 +91,108 @@ def __call__(self, event: dict, context: "LambdaContext") -> dict:
                 )
                 stack.enter_context(lifespan_cycle)
 
-            request_context = event["requestContext"]
+            scope = self.event_to_asgi_scope(event)
+            scope["aws.event"] = event
```
[the remark this answers is not in the document]
Comment (emcpow2) [reply mangum/adapter.py]: done

**Comment by emcpow2** [comment]
thanks. Done

**Comment by jordaneremieff** [comment]
Looks good, thanks!

**Pull request merged**